daemon/control.c

```diff
@@ -69,96 +71,71 @@ struct control {
-control_send_log_file(int fd, char *log_file_name, bool read_low_level, bool send_last_line_info)
+/**
+ * @brief 
+ * 
+ * @param fd: The file descriptor to send the logMessage protobuf message to 
+ * @param log_file_path the absolute path to the file to send
+ * @param log_file_name the file name only
+ */
+static void
+control_send_log_file(int fd, const char *log_file_path, char *log_file_name)
 {
-	int fp_low = -1;
-	bool skipped_lines = false;
-	FILE *fp = NULL;
-	char *line;
-	char line_low[LOGGER_ENTRY_MAX_LEN + 1];
-	size_t HEADER_LENGTH = 21;
-	ssize_t bytes_read;
-	bool file_is_open = false;
-
 	LogMessage message = LOG_MESSAGE__INIT;
-	message.prio = (LogPriority)LOGF_PRIO_DEBUG;
+
+	message.name = log_file_name;
 	DaemonToController out = DAEMON_TO_CONTROLLER__INIT;
 	out.code = DAEMON_TO_CONTROLLER__CODE__LOG_MESSAGE;
 	if (cmld_get_device_uuid()) {
 		TRACE("Setting uuid: %s", cmld_get_device_uuid());
 		out.device_uuid = mem_strdup(cmld_get_device_uuid());
 	}
 
-	DEBUG("Opening and sending %s", log_file_name);
-	if (read_low_level) {
-		fp_low = open(log_file_name, O_RDONLY | O_NONBLOCK);
-		if (fp_low == -1) {
-			ERROR("Could not open %s", log_file_name);
-		} else {
-			file_is_open = true;
-		}
-	} else {
-		fp = fopen(log_file_name, "r");
-		line = NULL;
-		if (fp == NULL) {
-			ERROR("Could not open %s", log_file_name);
-		} else {
-			file_is_open = true;
-		}
-	}
+	DEBUG("Opening and sending %s", log_file_path);
 
-	while (file_is_open) {
-		if (read_low_level) {
-			/* The driver let's us read entry by entry */
-			bytes_read = read(fp_low, line_low, LOGGER_ENTRY_MAX_LEN);
-			if (bytes_read <= 0)
-				break;
-			char *first_string = line_low + HEADER_LENGTH;
-			size_t string_len = strlen(first_string);
-			char *second_string = line_low + HEADER_LENGTH + string_len + 1;
-			message.msg = mem_printf("%s/%s", first_string, second_string);
-		} else {
-			size_t len = 0;
-			bytes_read = getline(&line, &len, fp);
-			if (bytes_read == -1)
-				break;
-			message.msg = mem_strdup(line);
-		}
-		out.log_message = &message;
-		if (protobuf_send_message(fd, (ProtobufCMessage *)&out) < 0) {
-			ERROR_ERRNO("Could not finish sending %s", log_file_name);
-			skipped_lines = true;
-			break;
-		}
-		mem_free0(message.msg);
+	off_t filesize = file_size(log_file_path);
+
+	char *file_buf = file_read_new(log_file_path, (size_t)filesize);
+
+	if (file_buf) {
+		message.msg = mem_strdup(file_buf);
+		free(file_buf);
+	} else {
+		DEBUG("File %s could not be read to buffer.", log_file_path);
 	}
-	if (send_last_line_info) {
-		message.msg = mem_printf("Last line of log");
-		out.log_message = &message;
-		if (protobuf_send_message(fd, (ProtobufCMessage *)&out) < 0) {
-			ERROR("Could not sent last line info for %s", log_file_name);
-		}
-		mem_free0(message.msg);
+	out.log_message = &message;
+	if (protobuf_send_message(fd, (ProtobufCMessage *)&out) < 0) {
+		ERROR_ERRNO("Could not finish sending %s", log_file_path);
```

Comment:
return error code in this case (define "ret" variable at function beginning and set to -1 here)